Marlin/src/module/temperature.cpp

```diff
@@ -951,15 +951,20 @@ void Temperature::manage_heater() {
   #endif
 
   #if HAS_THERMAL_PROTECTION
-    #if !defined(THERMAL_PROTECTION_GRACE_PERIOD)
-      #define THERMAL_PROTECTION_GRACE_PERIOD 0 // No grace period is needed for well-behaved boards.
+    #ifndef THERMAL_PROTECTION_GRACE_PERIOD
+      #define THERMAL_PROTECTION_GRACE_PERIOD 0 // No grace period needed on well-behaved boards
+    #endif
+    #if THERMAL_PROTECTION_GRACE_PERIOD > 0
+      static millis_t grace_period = ms + THERMAL_PROTECTION_GRACE_PERIOD;
+      if (ELAPSED(ms, grace_period)) grace_period = 0UL;
+    #else
+      static constexpr millis_t grace_period = 0UL;
     #endif
-    static millis_t grace_period = ms + THERMAL_PROTECTION_GRACE_PERIOD;
   #endif
 
   HOTEND_LOOP() {
     #if ENABLED(THERMAL_PROTECTION_HOTENDS)
-      if (ELAPSED(ms, grace_period) && degHotend(e) > temp_range[e].maxtemp)
+      if (!grace_period && degHotend(e) > temp_range[e].maxtemp)
```

Comment:
Looks nice :+1:

Reply:
This addition might be a good alternative to #13740, which applies a grace period to the mintemp/maxtemp kill behavior. Does it still look like both are needed?